Sources/Plasma/FeatureLib/pfPasswordStore/pfPasswordStore_Unix.cpp

```diff
@@ -43,16 +43,58 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 #include "pfPasswordStore.h"
 #include "pfPasswordStore_impl.h"
 
-#include "plProduct.h"
+#include "pnNetBase/pnNbSrvs.h"
 
+#include <string_theory/string>
+#include <libsecret/secret.h>
+
+static const SecretSchema pfPasswordStore_Schema = {
+    "com.mystonline.Password", SECRET_SCHEMA_NONE,
+    {
+        { "username", SECRET_SCHEMA_ATTRIBUTE_STRING },
+        { "server", SECRET_SCHEMA_ATTRIBUTE_STRING },
+        { nullptr, (SecretSchemaAttributeType)0 }
+    }
+};
 
 /*****************************************************************************
  ** pfUnixPasswordStore                                                     **
  *****************************************************************************/
 ST::string pfUnixPasswordStore::GetPassword(const ST::string& username)
 {
+    GError *error = nullptr;
+    gchar *password = secret_password_lookup_sync(&pfPasswordStore_Schema,
+                            nullptr, &error,
+                            "username", username.c_str(),
+                            "server", GetServerDisplayName().c_str(),
+                            nullptr);
+
+    ST::string result;
+    if (error) {
+        // Throw away the error and treat it as if no password was found...
+        g_error_free(error);
+    } else if (password) {
+        result = ST::string::from_utf8(password);
+        secret_password_free(password);
+    }
+
+    return result;
 }
 
-bool pfUnixPasswordStore::SetPassword(const ST::string& username, const ST::string& password)
+bool pfUnixPasswordStore::SetPassword(const ST::string& username,
+                                      const ST::string& password)
 {
+    GError *error = nullptr;
+    secret_password_store_sync(&pfPasswordStore_Schema, SECRET_COLLECTION_DEFAULT,
+                               "Myst Online Password", password.c_str(),
```

Comment:
Should the "Myst Online Password" include the server and/or username in the description?

The general pattern I see in Seahorse for Gnome apps is "[program name] password for [account ID]"